src/config.c

```diff
@@ -259,14 +261,19 @@ bool Config_ReadFromJSON(const char *cfg_data)
     READ_FLOAT(ui.bar_scale, DEFAULT_UI_SCALE);
     CLAMP(g_Config.ui.bar_scale, MIN_UI_SCALE, MAX_UI_SCALE);
 
-    struct json_array_s *layout_arr =
-        json_object_get_array(root_obj, "layout_sdl");
-    for (INPUT_ROLE role = 0; role < INPUT_ROLE_NUMBER_OF; role++) {
-        INPUT_SCANCODE scancode =
-            Input_GetAssignedScancode(INPUT_LAYOUT_USER, role);
-        scancode = json_array_get_int(layout_arr, role, scancode);
-        Input_AssignScancode(INPUT_LAYOUT_USER, role, scancode);
+    char *layout_name = Memory_Alloc(CONTROL_SCHEME_SIZE);
```

Comment:
No need to allocate it dynamically, just allocate it on stack.
`char layout_name[20];`
It's also OK to hardcode `CONTROL_SCHEME_SIZE` in this case, and I like to round the array sizes to increments of 10 hence `20` not `13`.